### ISSUE TITLE
Refactoring: findEnrichedSignatures() and sendToNetwork()

### ISSUE THREAD

**Issue by dongbohu** [greenelab/adage-server#346]
@mhuyck had the following comment when reviewing PR #345:

```
This method of invoking findEnrichedSignatures() leaks implementation details about the surrounding page (knowledge of $scope and $state) into what could be an independent component. I see that CommonGeneFuncts is used to dispatch the button press to the gene_network page in a similar fashion. This is apparently done to promote reuse of the sendToNetwork and findEnrichedSignatures helper functions, but the lack of clarity between what the method should know and what the surrounding page knows limits our ability to do any actual reuse.

I suppose this comment should be applied to an issue for future refactoring work since this will go beyond the scope of this PR (it includes modifying tribe_client), but the fix is fairly simple:

1. the code to pull the selectedGenes out of $scope and building the geneIds list should happen here
2. $scope and $state parameters should be replaced with geneIds and mlModel parameters
3. CommonGeneFuncts should inject $state as a dependency
4. both findEnrichedSignatures and sendToNetwork within CommonGeneFuncts should use the injected $state to dispatch to their corresponding pages

In other words, the common functions should be a thin wrapper around the $state.go utility and the parameters passed to them should mirror the parameters passed via the URL to the page.
```
